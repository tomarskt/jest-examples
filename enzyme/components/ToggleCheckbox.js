import React, { Component } from 'react';
import MyButton from './Mybutton'

class ToggleCheckbox extends Component {
  constructor(props) {
    super(props);
    this.state = { isChecked: false };
    this.onChange = this.onChange.bind(this);
  }

  onChange() {
    console.log(' ---------------------- this.onChange is invoked  ---------------------- ')
    this.setState({ isChecked: !this.state.isChecked });
  }

  render() {
    return (
      <div>
        <label>
          <input
            type="checkbox"
            checked={this.state.isChecked}
            onChange={this.onChange}
          />
          {this.state.isChecked ? 'On' : 'Off'}
        </label>
        <MyButton onClick={this.onChange} />
      </div>
    );
  }
}

export default ToggleCheckbox;