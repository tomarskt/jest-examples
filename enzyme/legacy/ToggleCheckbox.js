// Copyright 2004-present Facebook. All Rights Reserved.

import React from 'react';
import MyButton from './Mybutton'
export default class ToggleCheckbox extends React.Component {

  constructor(props) {
    super(props);
    this.state = { isChecked: false };

    this.onChange = this.onChange.bind(this);
  }

  componentDidMount() {
    console.log('######## componentDidMount called  ##################')
  }
  
  onChange() {
    this.setState({ isChecked: !this.state.isChecked });
  }

  render() {
    return (
      <div>
        <label>
          <input
            type="checkbox"
            checked={this.state.isChecked}
            onChange={this.onChange}
            />
          {this.state.isChecked ? 'On':'Off'}
        </label>
        <MyButton onClick={this.onChange} />
      </div>
    );
  }
}
