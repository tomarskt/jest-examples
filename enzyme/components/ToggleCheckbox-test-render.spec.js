import React from 'react';
import TestRenderer from 'react-test-renderer';
import ToggleCheckbox from './ToggleCheckbox';


describe('using react-test-renderer', () => {
    it('testRenderer', () => {
        const testRenderer = TestRenderer.create(<ToggleCheckbox />);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    })
});