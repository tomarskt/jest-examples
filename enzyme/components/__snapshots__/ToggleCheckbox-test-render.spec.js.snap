// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using react-test-renderer testRenderer 1`] = `
<div>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Off
  </label>
  <div>
    <button
      onClick={[Function]}
    >
      click me!
    </button>
  </div>
</div>
`;
